api/src/wfl/source.clj

```diff
@@ -289,62 +298,70 @@
     (when (seq shards->jobs)
-      (log/info (format "%s Snapshots created from new rows in %s.%s."
-                        (stage/log-prefix source) (:name dataset) table))
-      (write-snapshots-creation-jobs source now shards->jobs))
+      (log/info "Snapshots created from new rows."
+                :workload (workloads/to-log workload)
+                :dataset  (:name dataset)
+                :table    table)
+      (write-snapshots-creation-jobs workload now shards->jobs))
     ;; Even if our poll did not yield new rows to snapshot, at least we tried:
     (update-last-checked source now)))
 
 (defn ^:private update-pending-snapshot-jobs
   "Update the status of 'running' TDR snapshots in `source`."
-  [source]
-  (log/debug "Looking for running snapshot jobs..." :source source)
-  (let [pending-tdr-jobs (get-pending-tdr-jobs source)]
+  [workload]
+  (log/debug "Looking for running snapshot jobs."
+             :workload (workloads/to-log workload))
+  (let [pending-tdr-jobs (get-pending-tdr-jobs workload)]
     (when (seq pending-tdr-jobs)
       (->> pending-tdr-jobs
-           (map #(update % 1 check-tdr-job))
-           (run! #(write-snapshot-id source %)))
-      (log/debug "Running snapshot jobs updated." :source source))))
+           (map #(update % 1 (partial check-tdr-job workload)))
+           (run! #(write-snapshot-id workload %)))
+      (log/debug "Running snapshot jobs updated."
+                 :workload (workloads/to-log workload)))))
 
 ;; Workloads in general may be updated more frequently,
 ;; but overpolling the TDR increases the chances of:
 ;; - creating many single-row / low-cardinality snapshots
 ;; - locking the dataset / running into dataset locks
+;;
 (def ^:private tdr-source-default-polling-interval-minutes 20)
 
 (defn ^:private tdr-source-should-poll?
   "Return true if it's been at least the specified `polling_interval_minutes`
    or, if unspecified, `tdr-source-default-polling-interval-minutes`
    since `last_checked` -- when we last checked for new rows in the TDR."
-  [{:keys [type id last_checked pollingIntervalMinutes] :as _source} now]
+  [{{:keys [last_checked pollingIntervalMinutes]} :source :as workload} now]
   (let [checked            (timestamp-to-offsetdatetime last_checked)
         minutes-since-poll (.between ChronoUnit/MINUTES checked now)
         polling-interval   (or pollingIntervalMinutes
                                tdr-source-default-polling-interval-minutes)]
-    (log/debug {:type               type
-                :id                 id
-                :minutes-since-poll minutes-since-poll
-                :polling-interval   polling-interval})
+    (log/debug "Checking if we should poll TDR for new rows."
+               :workload           (workloads/to-log workload)
+               :minutes-since-poll minutes-since-poll
+               :polling-interval   polling-interval)
     (<= polling-interval minutes-since-poll)))
 
 (defn ^:private update-tdr-source
```

Comment:
Some of the logs emitted by this update loop:

```

{"message":{"column":5,"namespace":"wfl.source","result":"Checking if we should poll TDR for new rows.","polling-interval":1,"minutes-since-poll":1,"workload":{"uuid":"8c7dbd7a-1292-422a-95dc-b95f58fb0cb4","source":"TerraDataRepoSource","executor":"TerraExecutor","sink":"TerraDataRepoSink","labels":["hornet:test","project:(Test) okotsopo/GH-1617-workload-source-logs"]}},"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"file":"wfl/source.clj","function":"\"Checking if we should poll TDR for new rows.\"","line":348},"time":"2022-02-17T18:39:41.687384Z”}

{"message":{"column":3,"namespace":"wfl.source","result":"Looking for new rows in row metadata table","table":"inputs","interval":["2022-02-17T16:39:41","2022-02-17T18:39:41"],"workload":{"uuid":"8c7dbd7a-1292-422a-95dc-b95f58fb0cb4","source":"TerraDataRepoSource","executor":"TerraExecutor","sink":"TerraDataRepoSink","labels":["hornet:test","project:(Test) okotsopo/GH-1617-workload-source-logs"]},"dataset":"illumina_genotyping_arraye5e6b30f7ddc44e1a457c4bd9cbfc730"},"severity":"INFO","logging.googleapis.com/sourceLocation":{"file":"wfl/source.clj","function":"\"Looking for new rows in row metadata table\"","line":169},"time":"2022-02-17T18:39:41.710879Z”}

{"message":{"column":3,"namespace":"wfl.source","result":"Looking for running snapshot jobs.","workload":{"uuid":"8c7dbd7a-1292-422a-95dc-b95f58fb0cb4","source":"TerraDataRepoSource","executor":"TerraExecutor","sink":"TerraDataRepoSink","labels":["hornet:test","project:(Test) okotsopo/GH-1617-workload-source-logs"]}},"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"file":"wfl/source.clj","function":"\"Looking for running snapshot jobs.\"","line":322},"time":"2022-02-17T18:40:47.973148Z"}

{"message":{"column":7,"namespace":"wfl.source","result":"Running snapshot jobs updated.","workload":{"uuid":"8c7dbd7a-1292-422a-95dc-b95f58fb0cb4","source":"TerraDataRepoSource","executor":"TerraExecutor","sink":"TerraDataRepoSink","labels":["hornet:test","project:(Test) okotsopo/GH-1617-workload-source-logs"]}},"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"file":"wfl/source.clj","function":"\"Running snapshot jobs updated.\"","line":329},"time":"2022-02-17T18:40:50.343342Z"}
```